### ISSUE TITLE
ALB-128: Interleaved per-block optimizer causes training instability at peak LR

### ISSUE THREAD

**Issue by noahgift** [paiml/albor#113]
## Problem

entrenar's `CudaTransformerTrainer` updates each block's weights **during** the backward pass:

```
backward block 23 → optimizer step block 23 → backward block 22 → optimizer step block 22 → ...
```

PyTorch separates backward and optimizer:

```
backward block 23 → backward block 22 → ... → backward block 0 → optimizer step (all params)
```

With 24 blocks and peak LR 3e-4, the interleaved approach causes blocks to "fight" — block N-1's backward runs against stale weights from block N that were already updated in the same step.

## Evidence

v25 training (step 2K–6K val_ppl trajectory):

| Step | val_ppl | Scaling law prediction |
|------|---------|----------------------|
| 2K | **69.35** | 55.8 (optimistic) |
| 3K | 135.96 | 55.8 |
| 4K | **313.80** | 1113.1 (diverging!) |
| 5K | 138.13 | 502.0 |
| 6K | 131.98 | 313.2 |

ZClip detects gradient spikes z>2.0 every ~15 steps at peak LR. The step 4K spike coincides with the end of warmup where LR reaches its maximum.

## Five Whys

1. **Why does val_ppl spike at step 4K?** — Peak LR (3e-4) reached, maximum per-step weight perturbation.
2. **Why does peak LR cause instability here but not in PyTorch?** — Interleaved optimizer updates weights mid-backward.
3. **Why does interleaved update cause instability?** — Block N-1's backward uses block N's *updated* weights, not the weights that produced the forward activations.
4. **Why was the optimizer interleaved?** — VRAM constraint: storing all 24 blocks' gradients simultaneously requires ~2.4 GB more VRAM.
5. **Why not decouple now?** — Requires architectural refactor of `CudaTransformerTrainer::train_step()`.

## Options

1. **Reduce peak LR to 1.5e-4** (immediate, v26 config) — compensates for interleaved noise
2. **Decouple backward from optimizer** (future, needs +2.4 GB VRAM) — matches PyTorch behavior
3. **Gradient accumulation across blocks** (future) — average multiple micro-batches before each block update

## Contract

Related: C-BACKPARITY-001 (backward-parity-v1.yaml) — the interleaved optimizer is an architectural deviation from PyTorch's training loop that the backward parity contract does not currently cover.

## References

- v25 training log: `logs/v25-training.log`
- Spec: `docs/book/src/spec/06-training.md` §6.4.1
- entrenar `cuda_trainer.rs` `train_step()` method
